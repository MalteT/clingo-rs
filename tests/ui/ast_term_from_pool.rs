use clingo::*;

fn main() {
    let sym = Symbol::create_id("test", true).unwrap();
    let term1 = ast::Term::from(sym);
    let term2 = ast::Term::from(sym);
    let args = vec![term1,term2];
    let mut pool = ast::Pool::new(&args);
    let term3 = ast::Term::from(&pool);

    pool = ast::Pool::new(&args);
    drop(args);
    let _end= (pool, term3);
}