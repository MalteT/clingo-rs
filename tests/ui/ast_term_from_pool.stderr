error[E0506]: cannot assign to `pool` because it is borrowed
  --> $DIR/ast_term_from_pool.rs:11:5
   |
9  |     let term3 = ast::Term::from(&pool);
   |                                 ----- borrow of `pool` occurs here
10 | 
11 |     pool = ast::Pool::new(&args);
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ assignment to borrowed `pool` occurs here
12 |     drop(args);
13 |     let _end= (pool, term3);
   |                      ----- borrow later used here

error[E0505]: cannot move out of `args` because it is borrowed
  --> $DIR/ast_term_from_pool.rs:12:10
   |
8  |     let mut pool = ast::Pool::new(&args);
   |                                   ----- borrow of `args` occurs here
...
12 |     drop(args);
   |          ^^^^ move out of `args` occurs here
13 |     let _end= (pool, term3);
   |                ---- borrow later used here
